src/wallet/coin_selection.rs

```diff
@@ -242,6 +243,102 @@ impl<D: Database> CoinSelectionAlgorithm<D> for LargestFirstCoinSelection {
     }
 }
 
+/// Another simple and dumb coin selection
+///
+/// This coin selection algorithm sorts the available UTXOs by blockheight and then picks them starting
+/// from the oldest ones until the required amount is reached.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct OldestFirstCoinSelection;
+
+impl<D: Database> CoinSelectionAlgorithm<D> for OldestFirstCoinSelection {
+    fn coin_select(
+        &self,
+        database: &D,
+        required_utxos: Vec<WeightedUtxo>,
+        mut optional_utxos: Vec<WeightedUtxo>,
+        fee_rate: FeeRate,
+        amount_needed: u64,
+        mut fee_amount: u64,
+    ) -> Result<CoinSelectionResult, Error> {
+        use bitcoin::hash_types::Txid;
+
+        let mut txids: Vec<Txid> = optional_utxos
+            .iter()
+            .map(|wu| wu.utxo.outpoint().txid)
+            .collect::<Vec<Txid>>();
+
+        txids.dedup();
+
+        // query db and create a blockheight lookup table
+        //let blockheights: Result<HashMap<Txid, u32>, _> = txids
+        let mut blockheights: HashMap<Txid, u32> = HashMap::new();
+        for &txid in txids.iter() {
+            if let Some(TransactionDetails {
+                confirmation_time: Some(blocktime),
+                ..
+            }) = database.get_tx(&txid, false)?
+            {
+                blockheights.insert(txid, blocktime.height);
+            }
+        }
+
+        // We put the "required UTXOs" first and make sure the optional UTXOs are sorted from
+        // oldest to newest according to blocktime
+        // For utxo that doesn't exist in DB, they will have lowest priority to be selected
+        let utxos = {
+            optional_utxos.sort_unstable_by_key(|wu| {
+                blockheights.get(&wu.utxo.outpoint().txid).ok_or(u32::MAX)
+            });
+
+            println!("optional_utxos {:?}", optional_utxos);
+
+            required_utxos
+                .into_iter()
+                .map(|utxo| (true, utxo))
+                .chain(optional_utxos.into_iter().map(|utxo| (false, utxo)))
+        };
+
+        // Keep including inputs until we've got enough.
+        // Store the total input value in selected_amount and the total fee being paid in fee_amount
+        let mut selected_amount = 0;
+        let selected = utxos
+            .scan(
```

Comment:
this part of code is completely copy-and-paste from LargestFirstCoinSelection. I was trying to extract this logic to a new function `select_from_sorted_utxos` but found out that it's pretty hard to take a Iterator of struct as function argument. So I didn't continue on this one yet.

Reply:
Thank you for catching this. I have fixed it